spec/support/shared_examples/block_matcher.rb

```diff
@@ -0,0 +1,80 @@
+RSpec.shared_examples "an RSpec block-only matcher" do |*options|
+  # Note: Ruby 1.8 expects you to call a block with arguments if it is
+  # declared that accept arguments. In this case, some of the specs
+  # that include examples from this shared example group do not pass
+  # arguments. A workaround is to use splat and pick the first argument
+  # if it was passed.
+  options = options.first || {}
```

Comment:
Another approach would be to always add an empty hash as an argument when including this shared example group, both are workarounds, but this one is less scattered.

Had to use the same trick in `yield_spec.rb`.

Reply:
Can you not just do `|options = {}|` or does that not work on 1.8.7?

Reply:
Yes, the only reason for this is 1.8.7. Couldn't figure out a better approach.